### PR TITLE
QA-328

### PR DESCRIPTION
**Pull request by tbeckham**
First pass at adding the ability to specify a custom eucalyptus cookbook with euca-deploy.

### CODE REVIEW

File: bin/euca-deploy [hunk 1]
```diff
@@ -14,13 +14,15 @@ if __name__ == '__main__':
     parser.add_argument('-e', '--environment', default='etc/environment.yml')
     parser.add_argument('-p', '--password', default='foobar')
     parser.add_argument('-b', '--branch', default='euca-4.1')
+    parser.add_argument('-C', '--cookbook', default='https://github.com/eucalyptus/eucalyptus-cookbook')
```
Comment (viglesiasce) [inline bin/euca-deploy]: Id rather not have both -C and -c, also --cookbook is not descriptive. lets make it only --cookbook-repo, that way argparse will still work with --cookbook but the help will be more clear

File: bin/euca-deploy [hunk 2]
```diff
@@ -14,13 +14,15 @@ if __name__ == '__main__':
...
     parser.add_argument('--debug', action='store_true', default=False)
     args = parser.parse_args()
     component_deployer = ComponentDeployer(args.password,
                                            args.environment,
                                            args.config,
                                            args.debug,
-                                           args.branch)
+                                           args.branch,
+                                           args.cookbook)
```
Comment (viglesiasce) [inline bin/euca-deploy]: similarly change this to cookbook_repo

File: eucadeploy/componentdeployer.py [hunk 3]
```diff
@@ -23,9 +24,10 @@ def __init__(self, password, environment_file='environment.yml',
         with hide(*self.hidden_outputs):
             local('if [ ! -d eucalyptus-cookbook ]; then '
                   'git clone '
-                  'https://github.com/eucalyptus/eucalyptus-cookbook;'
-                  'fi')
+                  '{0} eucalyptus-cookbook;'
+                  'fi'.format(cookbook))
             local('cd eucalyptus-cookbook; git checkout {0};'.format(cookbook_branch))
+            local('git pull')
```
Comment (viglesiasce) [inline eucadeploy/componentdeployer.py]: i dont believe this will run in the right dir but could be mistaken

Comment (viglesiasce) [inline eucadeploy/componentdeployer.py]: We could use the fabric cd context manager: http://docs.fabfile.org/en/latest/api/core/context_managers.html#fabric.context_managers.cd

Comment (tbeckham) [inline eucadeploy/componentdeployer.py]: Nice catch, I incorrectly assumed that changing dir in the previous execute local line would persist.

**Comment by viglesiasce** [comment]
Looks good. Thanks @tbeckham.

**Pull request merged**